exploration/0002-expression-attributes.md

```diff
@@ -38,18 +38,70 @@ At least the following expression attributes should be considered:
 
   - `fallback` — A value to use instead of the default fallback,
     should the expression's primary formatting fail in some way.
+
+    > Example that would format as "Hello world" if `$place` is not defined:
+    >
+    > ```
+    > {Hello {$place @fallback=world}}
+    > ```
+
+  - `id` — An identifier for the expression.
+    This is included in the formatted part,
+    and allows the parts of an expression to be explicitly addressed.
+
+    > Example identifying two formatted numbers
+    > in a message with badly named input variables:
+    >
+    > ```
+    > {You received {$_0 :number @id=count} messages in {$_1 :number @id=time} hours.}
+    > ```
+
   - `locale` — An override for the locale used to format the expression.
     Should be expressed as a non-empty sequence of BCP 47 language codes.
+
+    > Example embedding a French literal in an English message:
+    >
+    > ```
+    > {In French, "{|bonjour| @locale=fr}" is a greeting}
+    > ```
+
   - `dir` — An override for the LTR/RTL/auto directionality of the expression.
 
+    > Example explicitly isolating the directionality of a placeholder:
+    >
+    > ```
+    > {Welcome, {$username @dir=auto}}
+    > ```
+
 - Attributes relevant for translators, tools, and other message operations,
   but with no runtime impact:
 
   - `example` — A literal value representing
     what the expression's formatted value will look like.
+
+    > Example providing a translator with an example for `$place`:
+    >
+    > ```
+    > {Hello {$place @example=world}}
+    > ```
+
   - `note` — A comment on the expression for translators.
+
+    > Example providing a translator with an example for `$place`:
+    >
+    > ```
+    > {Welcome to {$place @note=|The user's current location|}}
+    > ```
+
   - `translate` — A boolean `yes`/`no` indicator communicating to translators
     whether the expression should or should not be localised.
+
+    > Example embedding a non-translatable French literal in an English message:
+    >
+    > ```
+    > {In French, "{|bonjour| @locale=fr @translate=no}" is a greeting}
```

Comment:
See above: I like `bonjour` being part of the string better than as a placeholder operand. Placeholders often look like a little colored box and the translator has to expand the thing or carefully read the attributes to learn what it says.

Reply:
In this case, would that not be an appropriate representation of a non-translatable literal?